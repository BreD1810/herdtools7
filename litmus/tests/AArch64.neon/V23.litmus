AArch64 V23
(* Tests ST2 multiple structure *)
{
uint128_t b;
0:X1=x;0:X2=y;0:X3=z;0:X4=a;0:X0=b;
}
 P0                     ;
 MOVI V0.2S,#1          ;
 MOVI V1.2S,#1          ;
 ST2 {V0.2S,V1.2S},[X0] ;
 LD1 {V2.4S},[X0]       ;
 ST1 {V2.S}[0],[X1]     ;
 ST1 {V2.S}[1],[X2]     ;
 ST1 {V2.S}[2],[X3]     ;
 ST1 {V2.S}[3],[X4]     ;
forall(x=1 /\ y=1 /\ z=1 /\ a=1)
