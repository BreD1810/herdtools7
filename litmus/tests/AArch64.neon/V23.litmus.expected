Test V23 Required
Histogram (1 states)
1000000:>a=1; x=1; y=1; z=1;
Ok

Witnesses
Positive: 1000000, Negative: 0
Condition forall (x=1 /\ y=1 /\ z=1 /\ a=1) is validated
Hash=cbf99836a2bd1a89d2c8d3cee063bc3e
Observation V23 Always 1000000 0
